.github/workflows/rust.yml

```
@@ -16,51 +32,98 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/cache@v3
         with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-            target
+          path: ${{ env.CACHE_PATHS }}
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
       - name: Check source is valid
         run: cargo check --workspace
 
   formating:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Check formatting with rustfmt
+
+      - name: Check Rust formatting with rustfmt
         run: cargo fmt --all --check
 
+      - uses: brndnmtthws/rust-action-cargo-binstall@v1
+        with:
+          packages: taplo-cli
+
+      - name: Check TOML formatting with taplo
+        run: |
+          taplo fmt --check **/*/Cargo.toml
+
   tests:
     needs: validity
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Run all tests
-        run: cargo test --workspace --verbose --all-features
+      - uses: actions/cache@v3
+        with:
+          path: ${{ env.CACHE_PATHS }}
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            parser:
+              - 'parser/**'
+
+      - name: Run parser tests
+        if: steps.changes.outputs.parser == 'true'
+        run: cargo test
+        working-directory: parser
+
+      - name: Run base tests
+        run: cargo test
 
   fuzzing:
     needs: validity
     runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      matrix:
+        fuzz-type: [module_roundtrip_naive, windows-latest]
```

```suggestion
        fuzz-type: [module_roundtrip_naive, module_roundtrip_structured]
```

I would also maybe rename that to `fuzz-target` since that's more accurate once more functions are fuzzed

---

Definitely wasn't copying from the GH actions matrix documentation 🙈. `fuzz-target` is a much better name, I wasn't too confident with `fuzz-type`. Will change